### DISCUSSION TITLE
no trusted RSA public key found for 'alice@strongswan.org'

### DISCUSSION THREAD

**Discussion by deeptishm** [strongswan/strongswan#1171] [Q&A] [answered]
Getting Error no trusted RSA public key found for 'alice@strongswan.org' while  testing update_sa using https://www.strongswan.org/testing/testresults/ikev2/mobike-virtual-ip-nat/index.html testbed.

Sun Configuration:
`stack@stack-Standard-PC-i440FX-PIIX-1996:~$ sudo cat /etc/swanctl/swanctl.conf
connections {

   mobike {
      local_addrs  = 190.168.100.198
      remote_addrs  = %any
      pools = mobike

      local {
         auth = pubkey
         certs = sunCert.pem
         id = sun.strongswan.org
      }
      remote {
         auth = pubkey
         id = alice@strongswan.org
      }
      children {
         mobike {
            local_ts = 10.2.0.0/24
         }
      }
   }
}

pools {
   mobike {
      addrs = 10.3.0.3/32
   }
}`

`stack@stack-Standard-PC-i440FX-PIIX-1996:~$ sudo swanctl --load-creds
loaded certificate from '/etc/swanctl/x509/aliceCert.pem'
loaded certificate from '/etc/swanctl/x509/sunCert.pem'
loaded certificate from '/etc/swanctl/x509ca/strongswanCert.pem'
loaded certificate from '/etc/swanctl/x509crl/strongswan.crl'
loaded RSA key from '/etc/swanctl/private/sunKey.pem'
stack@stack-Standard-PC-i440FX-PIIX-1996:~$
stack@stack-Standard-PC-i440FX-PIIX-1996:~$ sudo swanctl --load-all
loaded certificate from '/etc/swanctl/x509/aliceCert.pem'
loaded certificate from '/etc/swanctl/x509/sunCert.pem'
loaded certificate from '/etc/swanctl/x509ca/strongswanCert.pem'
loaded certificate from '/etc/swanctl/x509crl/strongswan.crl'
loaded RSA key from '/etc/swanctl/private/sunKey.pem'
no authorities found, 0 unloaded
loaded pool 'mobike'
successfully loaded 1 pools, 0 unloaded
loaded connection 'mobike'
successfully loaded 1 connections, 0 unloaded
stack@stack-Standard-PC-i440FX-PIIX-1996:~$`

`stack@stack-Standard-PC-i440FX-PIIX-1996:~$ sudo swanctl --list-certs

List of X.509 End Entity Certificates

  subject:  "C=CH, O=strongswan, CN=sun.strongswan.org"
  issuer:   "C=CH, O=strongSwan, CN=strongSwan Root CA"
  validity:  not before Jul 25 13:42:02 2022, ok
             not after  Jul 25 13:42:02 2027, ok (expires in 1825 days)
  serial:    01
  altNames:  sun.strongswan.org
  flags:
  authkeyId: 4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75
  subjkeyId: e5:36:86:83:ab:be:c8:af:1e:e7:79:c5:23:f5:6f:3a:91:60:5a:93
  pubkey:    RSA 2048 bits, has private key
  keyid:     ba:5c:1c:c0:02:7e:3b:55:af:fb:8d:18:3b:23:d8:bc:91:7d:70:80
  subjkey:   e5:36:86:83:ab:be:c8:af:1e:e7:79:c5:23:f5:6f:3a:91:60:5a:93

  subject:  "C=CH, O=strongswan, CN=alice.strongswan.org"
  issuer:   "C=CH, O=strongSwan, CN=strongSwan Root CA"
  validity:  not before Jul 25 13:44:24 2022, ok
             not after  Jul 25 13:44:24 2027, ok (expires in 1825 days)
  serial:    01
  altNames:  alice.strongswan.org
  flags:
  authkeyId: 4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75
  subjkeyId: 08:f6:c3:81:8c:9b:02:e4:67:0a:ba:a4:83:54:bd:19:d2:6e:af:f3
  pubkey:    RSA 2048 bits
  keyid:     0a:fb:57:73:0f:e2:2c:2c:f4:f5:5d:e4:4b:ce:ea:e2:80:c3:1d:32
  subjkey:   08:f6:c3:81:8c:9b:02:e4:67:0a:ba:a4:83:54:bd:19:d2:6e:af:f3

List of X.509 CA Certificates

  subject:  "C=CH, O=strongSwan, CN=strongSwan Root CA"
  issuer:   "C=CH, O=strongSwan, CN=strongSwan Root CA"
  validity:  not before Jul 25 13:38:07 2022, ok
             not after  Jul 24 13:38:07 2032, ok (expires in 3651 days)
  serial:    33:8a:27:c5:bd:a2:78:5e
  flags:     CA CRLSign self-signed
  subjkeyId: 4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75
  pubkey:    RSA 2048 bits
  keyid:     c3:4a:83:92:b1:63:b6:a8:b5:4c:c4:e3:79:d7:d3:c0:1b:68:7c:37
  subjkey:   4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75

List of X.509 CRLs

  issuer:   "C=CH, O=strongSwan, CN=strongSwan Root CA"
  update:    this on Jul 25 15:15:32 2022, ok
             next on Aug 24 15:15:32 2022, ok (expires in 29 days)
  serial:    01
  authKeyId: 4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75
  0 revoked certificates
stack@stack-Standard-PC-i440FX-PIIX-1996:~$`

Alice Configuration:
`stack@stack-Standard-PC-i440FX-PIIX-1996:~$ sudo cat /etc/swanctl/swanctl.conf
connections {

   mobike {
      remote_addrs = 190.168.100.198

      vips = 0.0.0.0

      local {
         auth = pubkey
         certs = aliceCert.pem
         id = alice@strongswan.org
      }
      remote {
         auth = pubkey
         id = sun.strongswan.org
      }
      children {
         mobike {
            remote_ts = 10.2.0.0/24
         }
      }
   }
}
stack@stack-Standard-PC-i440FX-PIIX-1996:~$
`stack@stack-Standard-PC-i440FX-PIIX-1996:~$ sudo swanctl --load-creds
loaded certificate from '/etc/swanctl/x509/aliceCert.pem'
loaded certificate from '/etc/swanctl/x509/sunCert.pem'
loaded certificate from '/etc/swanctl/x509ca/strongswanCert.pem'
loaded certificate from '/etc/swanctl/x509crl/strongswan.crl'
loaded RSA key from '/etc/swanctl/private/aliceKey.pem'
stack@stack-Standard-PC-i440FX-PIIX-1996:~$

stack@stack-Standard-PC-i440FX-PIIX-1996:~$ sudo swanctl --load-all
loaded certificate from '/etc/swanctl/x509/aliceCert.pem'
loaded certificate from '/etc/swanctl/x509/sunCert.pem'
loaded certificate from '/etc/swanctl/x509ca/strongswanCert.pem'
loaded certificate from '/etc/swanctl/x509crl/strongswan.crl'
loaded RSA key from '/etc/swanctl/private/aliceKey.pem'
no authorities found, 0 unloaded
no pools found, 0 unloaded
loaded connection 'mobike'
successfully loaded 1 connections, 0 unloaded
stack@stack-Standard-PC-i440FX-PIIX-1996:~$`

stack@stack-Standard-PC-i440FX-PIIX-1996:~$ sudo swanctl --list-certs

List of X.509 End Entity Certificates

  subject:  "C=CH, O=strongswan, CN=alice.strongswan.org"
  issuer:   "C=CH, O=strongSwan, CN=strongSwan Root CA"
  validity:  not before Jul 25 13:44:24 2022, ok
             not after  Jul 25 13:44:24 2027, ok (expires in 1825 days)
  serial:    01
  altNames:  alice.strongswan.org
  flags:
  authkeyId: 4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75
  subjkeyId: 08:f6:c3:81:8c:9b:02:e4:67:0a:ba:a4:83:54:bd:19:d2:6e:af:f3
  pubkey:    RSA 2048 bits, has private key
  keyid:     0a:fb:57:73:0f:e2:2c:2c:f4:f5:5d:e4:4b:ce:ea:e2:80:c3:1d:32
  subjkey:   08:f6:c3:81:8c:9b:02:e4:67:0a:ba:a4:83:54:bd:19:d2:6e:af:f3

  subject:  "C=CH, O=strongswan, CN=sun.strongswan.org"
  issuer:   "C=CH, O=strongSwan, CN=strongSwan Root CA"
  validity:  not before Jul 25 13:42:02 2022, ok
             not after  Jul 25 13:42:02 2027, ok (expires in 1825 days)
  serial:    01
  altNames:  sun.strongswan.org
  flags:
  authkeyId: 4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75
  subjkeyId: e5:36:86:83:ab:be:c8:af:1e:e7:79:c5:23:f5:6f:3a:91:60:5a:93
  pubkey:    RSA 2048 bits
  keyid:     ba:5c:1c:c0:02:7e:3b:55:af:fb:8d:18:3b:23:d8:bc:91:7d:70:80
  subjkey:   e5:36:86:83:ab:be:c8:af:1e:e7:79:c5:23:f5:6f:3a:91:60:5a:93

List of X.509 CA Certificates

  subject:  "C=CH, O=strongSwan, CN=strongSwan Root CA"
  issuer:   "C=CH, O=strongSwan, CN=strongSwan Root CA"
  validity:  not before Jul 25 13:38:07 2022, ok
             not after  Jul 24 13:38:07 2032, ok (expires in 3651 days)
  serial:    33:8a:27:c5:bd:a2:78:5e
  flags:     CA CRLSign self-signed
  subjkeyId: 4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75
  pubkey:    RSA 2048 bits
  keyid:     c3:4a:83:92:b1:63:b6:a8:b5:4c:c4:e3:79:d7:d3:c0:1b:68:7c:37
  subjkey:   4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75

  subject:  "C=CH, O=strongSwan, CN=strongSwan Root CA"
  issuer:   "C=CH, O=strongSwan, CN=strongSwan Root CA"
  validity:  not before Jul 22 16:35:41 2022, ok
             not after  Jul 21 16:35:41 2032, ok (expires in 3648 days)
  serial:    49:91:9a:b9:e9:c8:dc:c5
  flags:     CA CRLSign self-signed
  subjkeyId: 16:93:46:22:77:97:de:1a:81:78:96:f7:fd:48:49:ed:85:7e:a5:90
  pubkey:    RSA 2048 bits
  keyid:     be:16:41:40:15:c0:08:e9:f6:00:70:5f:1d:6c:dd:45:1d:45:48:bb
  subjkey:   16:93:46:22:77:97:de:1a:81:78:96:f7:fd:48:49:ed:85:7e:a5:90

List of X.509 CRLs

  issuer:   "C=CH, O=strongSwan, CN=strongSwan Root CA"
  update:    this on Jul 25 15:15:32 2022, ok
             next on Aug 24 15:15:32 2022, ok (expires in 29 days)
  serial:    01
  authKeyId: 4c:1d:72:5d:68:e3:74:27:61:1f:f4:1b:ab:61:50:c7:aa:e2:1a:75
  0 revoked certificates
stack@stack-Standard-PC-i440FX-PIIX-1996:~$
`

Alice Initiates Connection
`stack@stack-Standard-PC-i440FX-PIIX-1996:~$ sudo swanctl --initiate --child mobike
[IKE] initiating IKE_SA mobike[16] to 190.168.100.198
[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
[NET] sending packet: from 190.168.100.174[500] to 190.168.100.198[500] (868 bytes)
[NET] received packet: from 190.168.100.198[500] to 190.168.100.174[500] (273 bytes)
[ENC] parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
[CFG] selected proposal: IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/CURVE_25519
[IKE] remote host is behind NAT
[IKE] received cert request for "C=CH, O=strongSwan, CN=strongSwan Root CA"
[IKE] sending cert request for "C=CH, O=strongSwan, CN=strongSwan Root CA"
[IKE] sending cert request for "C=CH, O=strongSwan, CN=strongSwan Root CA"
[IKE] authentication of 'alice@strongswan.org' (myself) with RSA_EMSA_PKCS1_SHA2_256 successful
[IKE] sending end entity cert "C=CH, O=strongswan, CN=alice.strongswan.org"
[IKE] establishing CHILD_SA mobike{16}
[ENC] generating IKE_AUTH request 1 [ IDi CERT N(INIT_CONTACT) CERTREQ IDr AUTH CPRQ(ADDR DNS) SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
[ENC] splitting IKE message (1568 bytes) into 2 fragments
[ENC] generating IKE_AUTH request 1 [ EF(1/2) ]
[ENC] generating IKE_AUTH request 1 [ EF(2/2) ]
[NET] sending packet: from 190.168.100.174[4500] to 190.168.100.198[4500] (1236 bytes)
[NET] sending packet: from 190.168.100.174[4500] to 190.168.100.198[4500] (404 bytes)
[NET] received packet: from 190.168.100.198[4500] to 190.168.100.174[4500] (80 bytes)
[ENC] parsed IKE_AUTH response 1 [ N(AUTH_FAILED) ]
[IKE] received AUTHENTICATION_FAILED notify error
initiate failed: establishing CHILD_SA 'mobike' failed
stack@stack-Standard-PC-i440FX-PIIX-1996:~$`

On Sun: /var/log/charon.log

`tack@stack-Standard-PC-i440FX-PIIX-1996:~$ tail -f /var/log/charon.log
Jul 25 22:04:22 09[LIB] <16> builder L3 CRED_PUBLIC_KEY - RSA of plugin 'pgp'
Jul 25 22:04:22 09[LIB] <16> builder L3 CRED_PUBLIC_KEY - RSA of plugin 'dnskey'
Jul 25 22:04:22 09[LIB] <16> builder L3 CRED_PUBLIC_KEY - RSA of plugin 'openssl'
Jul 25 22:04:22 09[IKE] <16> received end entity cert "C=CH, O=strongswan, CN=alice.strongswan.org"
Jul 25 22:04:22 09[CFG] <16> looking for peer configs matching 190.168.100.198[sun.strongswan.org]...190.168.100.174[alice@strongswan.org]
Jul 25 22:04:22 09[CFG] <mobike|16> selected peer config 'mobike'
**Jul 25 22:04:22 09[IKE] <mobike|16> no trusted RSA public key found for 'alice@strongswan.org'**
Jul 25 22:04:22 09[IKE] <mobike|16> peer supports MOBIKE
Jul 25 22:04:22 09[ENC] <mobike|16> generating IKE_AUTH response 1 [ N(AUTH_FAILED) ]
Jul 25 22:04:22 09[NET] <mobike|16> sending packet: from 190.168.100.198[4500] to 190.168.100.174[4500] (80 bytes)
`

**Comment by strongX509** [comment, marked as answer]
Well the error is quite trivial: you define Alice's ID as
````
 id = alice@strongswan.org
````
but the *subjectAltName* in Alice's certificate is
````
altNames: alice.strongswan.org
````
Thus correct either the *subjectAltName* in the certificate or the IKEv2 ID in the connection definition

**Reply by deeptishm** [reply]
Thanks for the response.. sorry for the inconvenience.
